pkg/http/monitoring.go

```diff
@@ -0,0 +1,48 @@
+package http
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/sirupsen/logrus"
+)
+
+// ListenAndServeMonitoring starts up an HTTP server serving /metrics and /health.
+//
+// When the context is canceled, the server will be gracefully shutdown.
+func ListenAndServeMonitoring(ctx context.Context, addr string, healthHandler http.Handler) error {
```

Comment:
how can we test it? would be nice to have a unit test that proves this thing running.